src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -800,15 +838,193 @@ def __try_add_error(error_list, detail):
+                    truncated_substatus_msg=self.__installation_substatus_msg_copy, substatus_index=installation_substatus_index)
+
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(json.dumps(truncated_status_file))
+            status_file_agent_size_diff = status_file_size_in_bytes - self.__internal_file_capacity
+            size_of_max_packages_allowed_in_status -= status_file_agent_size_diff   # Reduce the max packages byte size by tombstone, new error, and escape chars byte size
+
+        self.composite_logger.log_debug("End package list truncation")
+
+        return truncated_status_file
+
+    def __split_assessment_list(self, assessment_packages):
+        """ Split package list, keep 5 minimum packages, and remaining packages for truncation """
+        min_packages_count = Constants.StatusTruncationConfig.MIN_ASSESSMENT_PACKAGE_TO_RETAIN
+        min_assessment_patches_to_retain, assessment_patches_eligible_for_truncation = (assessment_packages[:min_packages_count], assessment_packages[min_packages_count:]) \
+            if len(assessment_packages) > min_packages_count else (assessment_packages, [])
+
+        return min_assessment_patches_to_retain, assessment_patches_eligible_for_truncation
+
+    def __apply_truncation_process(self, assessment_packages, installation_packages, max_package_list_capacity, low_pri_index=None):
+        """ Truncation function call split assessment method and apply truncation on assessment and installation packages """
+        installation_low_pri = []
+        installation_high_pri = installation_packages
+        # Cut assessment list into [:5], [5:]
+        min_assessment_patches_to_retain, assessment_patches_eligible_for_truncation = self.__split_assessment_list(assessment_packages)
+
+        if len(min_assessment_patches_to_retain) > 0:
+            max_package_list_capacity = max_package_list_capacity - self.__calc_package_payload_size_on_disk(min_assessment_patches_to_retain)
+
+        # Apply high priority (Failed, Installed) and low priority (Pending, Excluded, Not_Selected) installation logic, and keep min 5 assessment packages
+        if low_pri_index:
+            installation_high_pri = installation_packages[:low_pri_index]
+            installation_low_pri = installation_packages[low_pri_index:]
+
+        packages_retained_in_install_high_pri, packages_removed_from_inst_high_pri, remaining_list_capacity = self.__apply_truncation(installation_high_pri, max_package_list_capacity)
+        packages_retained_in_assessment, packages_removed_from_assessment, remaining_list_capacity = self.__apply_truncation(assessment_patches_eligible_for_truncation, remaining_list_capacity)
+        packages_retained_in_install_low_pri, packages_removed_from_inst_low_pri, _ = self.__apply_truncation(installation_low_pri, remaining_list_capacity)
```

Comment:
why not capturing remaining_capacity here?

Reply:
at line 978, this is the last truncation step, it's pointless to capture any remaining_capacity because we have retained as many "non-important low pri"packages as we can, and there's no packages that are less important than low pri packages to remove,

Reply:
@rane-rajasi to resolve if satisfied. Looks okay to me.